### PR TITLE
Fix System.IO.FileSystem.Tests

### LINKED ISSUE #3154: System.IO.FileSystem.Tests leaves behind temp directories that are nearly impossible for mortals to delete.
Repro:
  1. msbuild System.IO.FileSystem.Tests.csproj -t:test
  2. Try to clean out your %TMP% directory.
  3. Watch every normal method of deletion fail due to S.I.F.T. leaving behind massively subnested GUID-named directory trees that purposefully blow past the MAXPATH restriction.

I ended up having to write code to get rid of this stuff.

S.I.F.T. should clean up after itself.

### PR DESCRIPTION
**Pull request by Maxwe11**
Fixes #3154
/cc: @JeremyKuhne

### CODE REVIEW

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 1]
```diff
@@ -59,8 +59,17 @@ public void PathAlreadyExistsAsFile()
         public void PathAlreadyExistsAsDirectory(FileAttributes attributes)
         {
             DirectoryInfo testDir = Create(GetTestFilePath());
-            testDir.Attributes = attributes;
-            Assert.Equal(testDir.FullName, Create(testDir.FullName).FullName);
+            FileAttributes origial = testDir.Attributes;
```
Comment (nguerrera) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: typo: s/original/original/

**Comment by nguerrera** [comment]
LGTM apart from the typo. Thanks.

**Comment by stephentoub** [comment]
Thanks for fixing this. LGTM.

**Comment by Maxwe11** [comment]
CI completed

**Pull request merged**